packages/store/addon/-private/system/store.ts

```diff
@@ -2977,8 +3148,9 @@ class Store extends Service {
     super.willDestroy();
     this.recordArrayManager.destroy();
 
-    this._adapterCache = null;
-    this._serializerCache = null;
+    // Check if we need to null this out
```

Comment:
I *think* it is safe. Let's chat with @mike-north about how to cast things during destroy. This mattered more before we did `assertDestroyedStore` but as long as we are careful to keep our public API asserted in that way this cleanup is probably unnecessary.